### DISCUSSION TITLE
Why not support source within the code block itself? source: self

### DISCUSSION THREAD

**Discussion by congwenma** [coddingtonbear/obsidian-csv-table#16] [General]

**Comment by coddingtonbear** [comment]
There're several reasons including:

1. We're parsing the code block as YAML.  CSV is not valid yaml; so we'd need to figure out some sane way of parsing part of the block as yaml and the other part as CSV.
2. That seems like a really niche feature.  Creating a separate file is a pretty low bar.
3. I'm not personally interested in implementing such a feature, and I'd need a really good justification for the complexity that would add even if somebody else were to do the implementation.
